test/e2e/e2e_suite_test.go

```diff
@@ -51,6 +53,7 @@ func init() {
 }
 
 func TestE2E(t *testing.T) {
+	ctrl.SetLogger(klog.Background())
```

Comment:
This is needed because of the last statement in the CAPI v1.5.0 [release notes](https://cluster-api.sigs.k8s.io/developer/providers/migrations/v1.4-to-v1.5#notes-about-the-controller-runtime-bump):

> Ensure that the e2e test suite is setting a logger (e.g. via `ctrl.SetLogger(klog.Background())` in `TestE2E`. Otherwise logs are not visible and controller-runtime will print a warning.